gen/src/generate/char_property/bsearch_set.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::BTreeSet;
+
+/// A simple range-collapsing binary search array slice.
```

Comment:
Yeah, with the simple bsearch function, the pair is easier. The other one requires the index-based bsearch. Since there's not much perf/memory gain either way, let's keep it simple, as you said.

But, just to clear, the list method gets `high + 1` in the list, not `high` itself.

And, that brings back a question I faced when rewriting the Python scripts: do we want to keep these ranges closed, or half-closed? Right now they are closed, which is similar to Unicode syntax, but not much Rustic. Changing it would make reading the Rust side of the codes easier, I guess. What do you think?